contracts/Registry.sol

```diff
@@ -76,7 +76,8 @@ contract Registry is IRegistry, ERC721Holder, ERC1155Receiver, ERC1155Holder {
         uint256[] memory lendAmount,
         uint8[] memory maxRentDuration,
         bytes4[] memory dailyRentPrice,
-        IResolver.PaymentToken[] memory paymentToken
+        IResolver.PaymentToken[] memory paymentToken,
+        bytes1[] memory automaticRenewal
```

Comment:
btw, this doesn't save any gas. Because it is still treated as bytes32 in memory. Try out this function

```
    function takeArgs(uint256[] memory a, bytes1[] memory automaticRenewal) public pure {
        return;
    }
```
in remix ide and inspect the memory just before it returns. So, I'd suggest we just use bool here for simplicity, for now.

Reply:
and maybe let's call this argument: `willAutoRenew`?

Reply:
also, shit I have all these as `memory` instead of `calldata`.

Reply:
Ok, renamed to `willAutoRenew` and switched from `bytes1` to `bool`.

Will make those calldatas

Reply:
Correction, cannot switch memory for calldata because it leads to a stack too deep. We _may_ be able to avoid this if we pass everything in as a structure. Regardless of what we decide on, this is a big refactor that should be done in a separate PR.